src/gha_runner/gh.py

```diff
@@ -81,6 +81,33 @@ def _do_request(self, func, endpoint, **kwargs):
         else:
             return resp.json()
 
+    def create_runner_tokens(self, count: int) -> list[str]:
+        """Generate registration tokens for GitHub Actions runners.
+        This can be removed if this is added into PyGitHub.
+
+        Parameters
+        ----------
+        count : int
+            The number of runner tokens to generate.
+        Returns
+        -------
+        list[str]
+            A list of runner registration tokens.
+        Raises
+        ------
+        TokenCreationError
+            If there is an error generating the tokens.
+
+        """
+        tokens = []
+        for _ in range(count):
+            try:
+                token = self.create_runner_token()
+                tokens.append(token)
+            except TokenRetrievalError as e:
+                raise TokenRetrievalError(f"Error creating runner tokens: {e}")
```

Comment:
what's the reason for modifying the raised error here? This just changes the error message; should the underlying error message be better?

Reply:
Still a little confused here: your explanation sounds like you're trying to catch an error and rewrap in another error type, which would make sense. But the code is catching `TokenRetrievalError` and re-raising the same error type, which effectively just rewrites the error message.

Looking at the broader context of the code, it looks like this is maybe a missed part of a refactor where `create_runner_tokens` (plural) originally directly called `self.post`, and then was changed to call `create_runner_token` (singular), which already has this wrapping of arbitrary errors into a `TokenRetrievalError`.

Right now, it looks like this would give the following error message:

`Error creating runner tokens: Error creating runner token: $ORIGINAL_ERROR`

The outer wrap comes from `create_runner_tokens`, the inner wrap from `create_runner_token`. I don't think this is what was intended.

Reply:
Oooooooh, I completely misread the code, you are correct. I do want to raise the same error but I don't need to rewrap it, you are correct. 

With this in mind, I see two possible solutions:

1. Create a new error type to handle the multiple case
2. Reuse the same error and don't wrap it in a try-except

Which do you think would be more helpful in this context?

Reply:
I think letting the error bubble up is probably best here. No need for try-except: The same error means the same thing. (And checking again, I think the error message is even pretty clear -- adding the 's' to 'token' in "Error creating runner token: " doesn't seem needed.)

Reply:
I have implemented not using a try-except in the following commit: [0053439](https://github.com/omsf-eco-infra/gha-runner/pull/6/commits/0053439c86a10c64dbd07450680ce9cdcb91d3ca)